src/transformers/training_args_tf.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Tuple
+
+from .file_utils import cached_property, is_tf_available, tf_required
+from .training_args import TrainingArguments
+
+
+logger = logging.getLogger(__name__)
+
+if is_tf_available():
+    import tensorflow as tf
+
+
+@dataclass
+class TFTrainingArguments(TrainingArguments):
+    optimizer_name: str = field(
+        default="adam",
+        metadata={
+            "help": 'Name of a Tensorflow optimizer among "adadelta, adagrad, adam, adamax, ftrl, nadam, rmsprop, sgd, adamw"'
+        },
+    )
+    mode: str = field(
+        default="sequence-classification",
+        metadata={"help": 'Type of task, one of "sequence-classification", "token-classification" '},
+    )
+    loss_name: str = field(
+        default="SparseCategoricalCrossentropy",
+        metadata={
+            "help": "Name of a Tensorflow loss. For the list see: https://www.tensorflow.org/api_docs/python/tf/keras/losses"
+        },
+    )
+    eval_steps: int = field(default=1000, metadata={"help": "Run an evaluation every X steps."})
+    debug: bool = field(
+        default=False, metadata={"help": "Activate the trace to record computation graphs and profiling information"}
+    )
+
+    @cached_property
+    @tf_required
+    def _setup_strategy(self) -> Tuple["tf.distribute.Strategy", int]:
+        logger.info("Tensorflow: setting up strategy")
+        gpus = tf.config.list_physical_devices("GPU")
+
+        if self.no_cuda:
+            strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+        else:
+            try:
+                tpu = tf.distribute.cluster_resolver.TPUClusterResolver()
+            except ValueError:
+                tpu = None
```

Comment:
Could we possibly add a `--no_tpu` flag as well, similar to the `--no_cuda`? Talked about it with @julien-c a few minutes ago, we're probably going to do it this way in the PyTorch trainer

Reply:
Of course it is possible. So I suppose it will be in the `trainer_args.py` files?

Reply:
Indeed, in the `TrainingArguments` class! The[ PyTorch TPU trainer](https://github.com/huggingface/transformers/pull/4146/files#diff-ef17e4d152412f77ce542cec19496f81R109) currently has a `--use_tpu` flag but we'll switch it to `--no_tpu` instead later today.